core/src/main/java/org/infinispan/notifications/impl/AbstractListenerImpl.java

```diff
@@ -143,7 +148,6 @@ public void removeListener(Object listener) {
    }
 
    private void removeListenerInvocation(Class<? extends Annotation> annotation, Object listener) {
-      if (listener == null) return;
```

Comment:
Could this cause a NPE at line 154?  Or the List returned from the collection is always empty in your case?